Patch stolen from Mageia Cauldron but had to be modified quite a bit.

Fixed for void by svenper@tuta.io

--- filters/karbon/pdf/SvgOutputDev.cpp
+++ filters/karbon/pdf/SvgOutputDev.cpp
@@ -405,7 +405,7 @@ void SvgOutputDev::drawString(GfxState * state, GooString * s)
     const char * p = s->c_str();
     int len = s->getLength();
     CharCode code;
-    Unicode *u = nullptr;
+    const Unicode *u = nullptr;
     int uLen;
     double dx, dy, originX, originY;
     while (len > 0) {
@@ -476,7 +476,7 @@
 
 void SvgOutputDev::drawImage(GfxState *state, Object */*ref*/, Stream *str,
                              int width, int height, GfxImageColorMap *colorMap,
-                             int *maskColors, bool /*inlineImg*/)
+                             const int *maskColors, bool /*inlineImg*/)
 {
     ImageStream * imgStr = new ImageStream(str, width, colorMap->getNumPixelComps(), colorMap->getBits());
     imgStr->reset();
@@ -545,7 +545,7 @@ void SvgOutputDev::drawImage(GfxState *state, Object */*ref*/, Stream *str,
 
 void SvgOutputDev::drawImage(GfxState *state, Object *ref, Stream *str,
                              int width, int height, GfxImageColorMap *colorMap,
-                             bool /*interpolate*/, int *maskColors, bool inlineImg)
+                             bool /*interpolate*/, const int *maskColors, bool inlineImg)
 {
     drawImage(state, ref, str, width, height, colorMap, maskColors, inlineImg);
 }
--- filters/karbon/pdf/SvgOutputDev.h
+++ filters/karbon/pdf/SvgOutputDev.h
@@ -63,10 +63,10 @@
     // images
     virtual void drawImage(GfxState *state, Object *ref, Stream *str,
                            int width, int height, GfxImageColorMap *colorMap,
-                           int *maskColors, bool inlineImg);
+                           const int *maskColors, bool inlineImg);
     virtual void drawImage(GfxState *state, Object *ref, Stream *str,
                            int width, int height, GfxImageColorMap *colorMap,
-                           bool interpolate, int *maskColors, bool inlineImg);
+                           bool interpolate, const int *maskColors, bool inlineImg);

     // styles
     virtual void updateAll(GfxState *state);
